### PR TITLE
[stable/prometheus-operator] add runbook_url prefix (#19810)

### LINKED ISSUE #19810: [stable/prometheus-operator] Allow defaultRules to configure runbook_url base path
**Is your feature request related to a problem? Please describe.**
The defaultRules from the upstream kubernetes-mixin are very useful but the default runbook_url points to an example repository.

**Describe the solution you'd like**
I'd like to just configure the first prefix of the url globally:
`runbook_url: https://github.com/kubernetes-monitoring/kubernetes-mixin/tree/master/runbook.md#alert-name-kubepersistentvolumeerrors`
Would become
`runbook_url: {{ .Values.defaultRules.runbook_url }}alert-name-kubepersistentvolumeerrors`

With the default value being
runbook_url: `https://github.com/kubernetes-monitoring/kubernetes-mixin/tree/master/runbook.md#`

**Describe alternatives you've considered**
There's potentially some more complex templating that could be done to build a URL but I don't think that's within the scope of this Helm chart.

### PR DESCRIPTION
**Pull request by seanson**
### What this PR does / why we need it:

Implements a new value config for prefixing runbook_url annotations in default rules.

#### Which issue this PR fixes
  - fixes #19810 

#### Checklist
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by bismarck** [comment]
@seanson The change looks good. Could you rerun the sync script and update your PR?

**Comment by seanson** [comment]
> @seanson The change looks good. Could you rerun the sync script and update your PR?

@bismarck - I've done a re-sync for every template except `templates/prometheus/rules-1.14/prometheus.yaml` which just has an upstream change to its description that's causing a linting error in the CI runs.

File: stable/prometheus-operator/hack/sync_prometheus_rules.py [hunk 1]
```diff
@@ -105,6 +105,9 @@ def new_representer(dumper, data):
     'alertmanager-$1': {
         'replacement': '$1',
         'init': ''},
+    'https://github.com/kubernetes-monitoring/kubernetes-mixin/tree/master/runbook.md#': {
```
Comment (bismarck) [inline stable/prometheus-operator/hack/sync_prometheus_rules.py:108]: Could you remove the hash from the URL for the replacement?  Please don't forget to update the README.

**Review by bismarck — CHANGES_REQUESTED** [review]

File: stable/prometheus-operator/hack/sync_prometheus_rules.py (hunk 1 above)
Comment (seanson) [reply stable/prometheus-operator/hack/sync_prometheus_rules.py:108]: I left the hash in so runbook prefixes could allow for several options:
`https://example.com/runbook#alert` for single page run books
`https://example.com/runbook/alert` for multiple page run books
`https://example.com/search?q=alert` an example prefix for searching something like confluence

If this is overly complex let me know and I'll remove the hash.

Comment (bismarck) [reply stable/prometheus-operator/hack/sync_prometheus_rules.py:108]: I think that's a reasonable decision.

**Comment by bismarck** [comment]
@seanson Could you bump the chart version again?

**Pull request merged**